marlowe/test/Spec/Marlowe/Semantics/Golden/Swap.hs

```diff
@@ -0,0 +1,108 @@
+              Close
+            ]
+            bTimeout
+            Close
+      ]
+      aTimeout
+      Close
+
+
+-- | A wrapper to assist parsing of test cases.
+newtype Map k v = Map {unMap :: [(k, v)]}
+
+
+-- | A function to assist parsing of test cases.
+toAM :: Map k v -> AM.Map k v
+toAM = AM.fromList . unMap
+
+
+-- An orphan instance to support parsing of test cases.
+instance IsString AccountId where
+  fromString ('P' : 'K' : x) = PK $ fromString x
+  fromString x               = Role $ fromString x
+
+
+-- | A list of Pangram test cases and results that should succeed.
+valids :: [(POSIXTime, [TransactionInput], TransactionOutput)]
+valids =
+  [
+    (POSIXTime {getPOSIXTime = 0}, [TransactionInput {txInterval = (POSIXTime {getPOSIXTime = 1000},POSIXTime {getPOSIXTime = 1000}), txInputs = []}], TransactionOutput {txOutWarnings = [], txOutPayments = [], txOutState = State {accounts = toAM $ Map {unMap = []}, choices = toAM $ Map {unMap = []}, boundValues = toAM $ Map {unMap = []}, minTime = POSIXTime {getPOSIXTime = 1000}}, txOutContract = Close})
+  , (POSIXTime {getPOSIXTime = 0}, [TransactionInput {txInterval = (POSIXTime {getPOSIXTime = 0},POSIXTime {getPOSIXTime = 0}), txInputs = [NormalInput (IDeposit "Party A" "Party A" (Token "13e78e78c233e131b0cbe4424225d338b7c5ac65e16df0a3e6c9d8f8" "Token A") 300000000)]},TransactionInput {txInterval = (POSIXTime {getPOSIXTime = 2000},POSIXTime {getPOSIXTime = 2000}), txInputs = []}], TransactionOutput {txOutWarnings = [], txOutPayments = [Payment "Party A" (Party "Party A") (Value(toAM $ Map [("13e78e78c233e131b0cbe4424225d338b7c5ac65e16df0a3e6c9d8f8", toAM $ Map [("Token A",300000000)])]))], txOutState = State {accounts = toAM $ Map {unMap = []}, choices = toAM $ Map {unMap = []}, boundValues = toAM $ Map {unMap = []}, minTime = POSIXTime {getPOSIXTime = 2000}}, txOutContract = Close})
+  , (POSIXTime {getPOSIXTime = 0}, [TransactionInput {txInterval = (POSIXTime {getPOSIXTime = 0},POSIXTime {getPOSIXTime = 0}), txInputs = [NormalInput (IDeposit "Party A" "Party A" (Token "13e78e78c233e131b0cbe4424225d338b7c5ac65e16df0a3e6c9d8f8" "Token A") 300000000)]},TransactionInput {txInterval = (POSIXTime {getPOSIXTime = 0},POSIXTime {getPOSIXTime = 0}), txInputs = [NormalInput (IDeposit "Party B" "Party B" (Token "1b9af43b0eaafc42dfaefbbf4e71437af45454c7292a6b6606363741" "Token B") 500000000)]}], TransactionOutput {txOutWarnings = [], txOutPayments = [Payment "Party A" (Party "Party B") (Value(toAM $ Map [("13e78e78c233e131b0cbe4424225d338b7c5ac65e16df0a3e6c9d8f8", toAM $ Map [("Token A",300000000)])])),Payment "Party B" (Party "Party A") (Value(toAM $ Map [("1b9af43b0eaafc42dfaefbbf4e71437af45454c7292a6b6606363741", toAM $ Map [("Token B",500000000)])]))], txOutState = State {accounts = toAM $ Map {unMap = []}, choices = toAM $ Map {unMap = []}, boundValues = toAM $ Map {unMap = []}, minTime = POSIXTime {getPOSIXTime = 0}}, txOutContract = Close})
+  ]
```

Comment:
These were automatically generated and weren't really meant to be human readable. Our `Show` instances don't all output parsable Haskell because the types shown don't export the constructors that are used by `show`. I was tempted to embed JSON (either directly or with the `file-embed` package). I'm adding a note to the documentation that these are machine generated.

Reply:
Fixed in 04f14adfbd2717698b65fa9f742f00792bb241fb.